src/homepage/_posts/2023-06-20-fail2ban.markdown

```diff
@@ -0,0 +1,356 @@
+ <PatternLayout pattern="${logLayoutJames}" charset="UTF-8" />
+ <Policies>
+ <SizeBasedTriggeringPolicy size="10 MB" />
+ </Policies>
+ <DefaultRolloverStrategy min="1" max="9" />
+ </RollingFile>
+ </Appenders>
+ <Loggers>
+ <Logger name="org.apache.james.protocols.smtp" level="info" >
+ <AppenderRef ref="James" level="info" />
+ </Logger>
+ <Root level="info" >
+ <AppenderRef ref="Console" level="info" />
+ </Root>
+ </Loggers>
+</Configuration>
+```
+
+To test the log output, we first fill the file (/opt/james/log/james.log) with the following content:
+
+```
+2023-06-17 11:11:50.206 +0200 -
+org.apache.james.protocols.smtp.core.fastfail.AbstractValidRcptHandler
+[ip=45.129.14.30] - Rejected message. Unknown user: dar...@domaine.de
+2023-06-17 11:11:51.206 +0200 -
+org.apache.james.protocols.smtp.core.fastfail.AbstractValidRcptHandler
+[ip=45.129.14.31] - Rejected message. Unknown user: dar...@domaine.de
+2023-06-17 11:11:50.206 +0200 -
+org.apache.james.protocols.smtp.core.esmtp.AuthCmdHandler [ip=] - AUTH method
+LOGIN failed from Username{localPart=monitor, domainPart=Optional[Domain :
+domaine.de]}@45.129.14.40
+2023-06-17 11:11:50.206 +0200 -
+org.apache.james.protocols.smtp.core.esmtp.AuthCmdHandler [ip=] - AUTH method
+LOGIN failed from Username{localPart=monitor, domainPart=Optional[Domain :
+domaine.de]}@45.129.14.41
+2023-06-17 11:11:52.206 +0200 -
+org.apache.james.protocols.smtp.core.fastfail.AbstractValidRcptHandler
+[ip=45.129.14.30] - Rejected message. Unknown user: dar...@domaine.de
+2023-06-17 11:11:53.206 +0200 -
+org.apache.james.protocols.smtp.core.fastfail.AbstractValidRcptHandler
+[ip=45.129.14.31] - Rejected message. Unknown user: dar...@domaine.de
+2023-06-17 11:11:50.206 +0200 -
+org.apache.james.protocols.smtp.core.esmtp.AuthCmdHandler [ip=] - AUTH method
+LOGIN failed from Username{localPart=monitor, domainPart=Optional[Domain :
+domaine.de]}@45.129.14.40
+2023-06-17 11:11:50.206 +0200 -
+org.apache.james.protocols.smtp.core.esmtp.AuthCmdHandler [ip=] - AUTH method
+LOGIN failed from Username{localPart=monitor, domainPart=Optional[Domain :
+domaine.de]}@45.129.14.41
+```
+
+## Setting up filters
+
+We create the file "/etc/fail2ban/filter.d/james.conf" with the following content:
+
+```
+[Definition]
+failregex =
+^.-.org.apache.james.protocols.smtp.core.fastfail.AbstractValidRcptHandler.\
+[ip=<HOST>\].-.Rejected message. Unknown user.*$
+ ^.-.org.apache.james.protocols.smtp.core.esmtp.AuthCmdHandler.\[ip=\].-.AUTH
+method LOGIN failed from Username.*\@<HOST>$
+ignoreregex =
+We see 2 regular expressions there. The first looks at the „AbstractValidRcptHandler“ class. The IP
+address is filled via the James variable. The IP address is recognized with the placeholder <HOST>.
+The second looks at the "AuthCmdHandler" class, the IP address is at the end of the line.
+Setup Jail
+We create the file "/etc/fail2ban/jail.d/james.conf" with the following content:
+[james]
```

Comment:
````suggestion
ignoreregex =
```
We see 2 regular expressions there. The first looks at the „AbstractValidRcptHandler“ class. The IP
address is filled via the James variable. The IP address is recognized with the placeholder <HOST>.

The second looks at the "AuthCmdHandler" class, the IP address is at the end of the line.

## Setup Jail

We create the file "/etc/fail2ban/jail.d/james.conf" with the following content:
```
[james]
````